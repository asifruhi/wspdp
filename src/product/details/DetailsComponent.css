.all-products-link {
	display: block;
	padding-bottom: 10px;
	text-align: center;
}

.product-detail-container {
	margin-top: 50px;
}

@media screen and (max-width: 35.5em) {
	.product-detail-container {
		margin-top: 20px;
	}
}

.product-detail-container .product-image {
	text-align: center;
	cursor: pointer;
}

.product-detail-container .product-detail {
	padding: 0 10px;
}

.product-detail-container .product-detail .product-price {
  font-weight: 500;
  line-height: 1.5em;
}

.product-detail-container .product-detail .regular {
	color: grey;
}

.product-detail-container .product-detail .selling {
	color: red;
	opacity: 0.8;
}

.slider-control-bottomcenter {
  background: rgba(192, 192, 192, 0.7);
  width: 100%;
  text-align: center;
}

@media screen and (max-width: 35.5em) {
	.images-overlay img {
		width: 30em;
		height: 30em;
	}

	.slider-control-bottomcenter {
		display: none;
  }
}

@media screen and (min-width: 35.5em) and (max-width: 48em) and (orientation: portrait) {
	.images-overlay img {
		width: 35.5em;
		height: 35.5em;
	}
}

@media screen and (min-width: 35.5em) and (max-width: 48em) and (orientation: landscape) {
	.images-overlay img {
		width: 18em!important;
		height: 18em;
		margin-left: 25%;
	}
}


@media screen and (min-width: 48em) and (max-width: 64em) and (orientation: portrait) {
	.images-overlay img {
		width: 48em;
		height: 48em;
	}
}

@media screen and (min-width: 48em) and (max-width: 64em) and (orientation: landscape) {
	.images-overlay img {
		width: 24em!important;
		height: 24em;
		margin-left: 25%;
	}
}


@media screen and (min-width: 64em) and (max-width: 80em) and (orientation: portrait) {
	.images-overlay img {
		width: 64em;
		height: 64em;
	}
}

@media screen and (min-width: 64em) and (max-width: 80em) and (orientation: landscape) {
	.images-overlay img {
		width: 32em!important;
		height: 32em;
		margin-left: 25%;
	}
}


@media screen and (min-width: 80em) and (orientation: portrait) {
	.images-overlay img {
		width: 80em;
		height: 80em;
	}
}

@media screen and (min-width: 80em) and (orientation: landscape) {
	.images-overlay img {
		width: 40em!important;
		height: 40em;
		margin-left: 25%;
	}
}


