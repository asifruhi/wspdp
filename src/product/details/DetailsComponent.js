import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { AllHtmlEntities } from 'html-entities';

import Carousel from 'nuka-carousel';

import priceFormat from '../../utils/priceFormat';
import Overlay from '../../common/Overlay';

import './DetailsComponent.css';

const entities = new AllHtmlEntities();

class ProductDetails extends React.Component {
	static propTypes = {
		product: PropTypes.object.isRequired,
		errors: PropTypes.array
	}

	static defaultProps = {
		errors: []
	}

	constructor(props) {
		super(props);

		this.showAltImages = this.showAltImages.bind(this);
	}

	showAltImages() {
		this.overlay.show();
	}

	render() {
		const { product } = this.props;
		return product ? (
			<div>
				<Link to='/products' className="all-products-link">View All Products</Link>
				<div className="pure-g product-detail-container" role="article">
					<div className="pure-u-1 pure-u-md-1-2 product-image">
            <img alt={product.name} src={product.hero.href} onClick={this.showAltImages} tabIndex="5"/>
          </div>
          <div className="pure-u-1 pure-u-md-1-2">
            <div className="product-detail">
		          <h1 tabIndex="2">{entities.decode(product.name)}</h1>
		          <div role="contentinfo">
		            {product.priceRange.regular ? (<div className="regular product-price" tabIndex="3">Regular Price: {priceFormat(product.priceRange.regular.low)} - {priceFormat(product.priceRange.regular.high)}</div>) : ''}
		            {product.priceRange.selling ? (<div className="selling product-price" tabIndex="4">Selling Price: {priceFormat(product.priceRange.selling.low)} - {priceFormat(product.priceRange.selling.high)}</div>) : ''}
		          </div>
		        </div>
	         </div>
        </div>
        <Overlay ref={el => this.overlay = el}>
					<Carousel className="images-overlay">
						<img key='img0' alt={product.name} src={product.hero.href}/>
						{
							product.images.map((image, idx) => <img alt={product.name} key={'img' + (idx + 1)} src={image.href}/>)
						}
					</Carousel>
        </Overlay>
			</div>
		) : (<div></div>);
	}
}

export default ProductDetails;