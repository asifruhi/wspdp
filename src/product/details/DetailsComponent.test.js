import React from 'react';
import renderer from 'react-test-renderer';

import DetailsComponent from './DetailsComponent';

it('renders product details correctly', () => {
	const product = {
		id: "metallic-square-lanterns-d5161",
		name: "Metallic Square Lanterns",
		priceRange:{
			regular: {high:149,low:24},
			selling:{high:111,low:18},
			type:"special"
		},
		hero:{
			href: "https://www.westelm.com/weimgs/ab/images/wcm/products/201810/0031/metallic-square-lanterns-m.jpg"
		},
		images:[{
			href: "https://www.westelm.com/weimgs/ab/images/wcm/products/201810/0031/metallic-square-lanterns-1-m.jpg"
		}, {
			href: "https://www.westelm.com/weimgs/ab/images/wcm/products/201811/0001/metallic-square-lanterns-m.jpg"
		}]
	};
  const tree = renderer
    .create(<DetailsComponent product={product}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});