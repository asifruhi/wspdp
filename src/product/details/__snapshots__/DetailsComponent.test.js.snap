// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders product details correctly 1`] = `
<div>
  <a
    href="/products"
    role="link"
  >
    View All Products
  </a>
  <div
    className="pure-g product-detail-container"
  >
    <div
      className="pure-u-1 pure-u-md-1-2 product-image"
    >
      <img
        alt="image for Metallic Square Lanterns"
        onClick={[Function]}
        src="https://www.westelm.com/weimgs/ab/images/wcm/products/201810/0031/metallic-square-lanterns-m.jpg"
        tabIndex="5"
      />
    </div>
    <div
      className="pure-u-1 pure-u-md-1-2"
    >
      <div
        className="product-detail"
      >
        <h1
          aria-label="Product name is Metallic Square Lanterns"
          tabIndex="2"
        >
          Metallic Square Lanterns
        </h1>
        <div>
          <div
            aria-label="Regular price range is $24 to $149"
            className="regular product-price"
            tabIndex="3"
          >
            Regular Price: 
            $24
             - 
            $149
          </div>
          <div
            aria-label="Selling price range is $18 to $111"
            className="selling product-price"
            tabIndex="4"
          >
            Selling Price: 
            $18
             - 
            $111
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
